### PR TITLE
Semantic Tasks 2.1.0 (KM-A)

### PR DESCRIPTION
**Pull request by thomas-topway-it**
This PR is made in reference to: [KM-A Knowledge Management Associates](https://km-a.net/knowledge-wiki/) upgrade of Semantic Tasks
This PR addresses or contains:
- fix issues/52
- fix issues/49
- fix issues/39
- fix issues/38

See https://github.com/Knowledge-Wiki/SemanticTasks/blob/master/docs/RELEASE-NOTES.md

This PR includes:
- [ ] Tests (unit/integration)
- [ ] CI build passed

### CODE REVIEW

**Comment by thomas-topway-it** [comment]
see also https://www.mediawiki.org/wiki/Extension:Semantic_Tasks

**Comment by krabina** [comment]
This PR fixes

#52 
#49 
#39 
#38

File: Assignees.php [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+use ParserOutput;
+use SMW\ApplicationFactory;
+use SMW\DIWikiPage;
+use SMWDataItem;
+use User;
+use WikiPage;
+
+/** @todo: rename TaskDiff something similar */
+class Assignees {
```
Comment (JeroenDeDauw) [inline Assignees.php:13]: Why did you copy this class?

Comment (thomas-topway-it) [reply Assignees.php]: hello Jeroen, in the root folder ? it was an error of github file upload, you can delete it

**Comment by krabina** [comment]
Anything we can do to get this accepted?

**Comment by krabina** [comment]
The file was already deleted by Thomas, so I guess this is ready for acceptance?

**Comment by gesinn-it-gea** [comment]
I suggest to add GitHub Actions CI and PHPStan to help with reviews. @Jereon: shall we merge anyhow already and ask @thomas-topway-it and @krabina to improve testing in a separate PR?

**Comment by Jereon** [comment]
Hello.  I want to ask a couple of questions.  Why am I tagged in this
email?  Am I related to your dialogue?

чт, 6 окт. 2022 г., 11:59 Bernhard Krabina ***@***.***>:

**Comment by JeroenDeDauw** [comment]
@Jereon they meant to ping me

@krabina well how do we know this code does what it is intended to do? how do we know it does not break key existing functionality? You can't reliably verify that just by looking at the code.

We could YOLO merge... Especially if you manually tested. Perhaps that is the best we can do for this project.

**Comment by krabina** [comment]
I guess since #52 is blocking usage in SMW, it is best for this repo to accept this. Or wait for others to test our fork? I have it in production in two wikis, one is still on SMW 3.x, the other in SMW 4.

We are willing to add testing in the future. Maybe even in the next weeks. Would be nice if someone who also uses this would test our version.

**Pull request merged**